src/modinfo.h

```diff
@@ -812,17 +815,22 @@ class ModInfo : public QObject, public MOBase::IModInterface
 
 protected:
 
+  /**
+   *
+   */
   ModInfo(PluginContainer *pluginContainer);
 
-  static void updateIndices();
-  static bool ByName(const ModInfo::Ptr &LHS, const ModInfo::Ptr &RHS);
-
   /**
-   * @brief check if the content of this mod is valid.
+   * @brief Prefetch content for this mod.
```

Comment:
The comment isn't really clear, maybe comment that it will ensure that the top leyer of the contentTree is populated and it's purpose is performance optimization.

Reply:
I we are caching, I'm probably gonna prefetch by actually caching `isValid` and `getContents` directly.

Reply:
So not storing the tree but just the results if isValid and getContents?
Wouldn't it be better to also store the tree for other stuff?

Reply:
The tree is already cached, which was the main thing I wanted to do.

Reply:
Also if you are prefetch querying the plugin and we are dealing with a python one, the actual plugin calls will get serialized loosing the advantage of paralleling loadfromdisk.

Reply:
I'll try to see what it does. I can still populate one level of the tree to get a bit of multi-threading on the C++ side.

Can't do much for the python stuff...

Reply:
I would just make it so that like the tree, isValid and getContents can also get invalidated by diskContentModified() and simply the next call to isValid will do the computation but all following calls will use the new cached value until it's invalidated again.